### PR TITLE
Propagate annotations from neo into SI object.

### PR DESCRIPTION
**Pull request by samuelgarcia**

### CODE REVIEW

File: spikeinterface/extractors/neoextractors/neobaseextractor.py [hunk 1]
```diff
@@ -76,6 +72,26 @@ def __init__(self, stream_id=None, **neo_kwargs):
...
+            # scalar annotations
+            for k, v in sig_ann.items():
+                if not k.startswith('__'):
+                    self.annotate(k=v)
+            # vector array_annotations are channel properties
+            for k, values in sig_ann['__array_annotations__'].items():
+                self.set_property(k, values)
+            
+            print(self._properties)
+            print(self._annotations)
```
Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/neobaseextractor.py:94]:
```suggestion
```

File: spikeinterface/extractors/neoextractors/blackrock.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ class BlackrockRecordingExtractor(NeoBaseRecordingExtractor):
     NeoRawIOClass = 'BlackrockRawIO'
 
     def __init__(self, file_path, stream_id=None):
-        neo_kwargs = {'filename': str(file_path)}
+        neo_kwargs = {'filename': str(file_path), 'all_annotations': True}
```
Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/blackrock.py:20]: Why is this `True` by default only for Blackrock?

**Comment by samuelgarcia** [comment]
it is now a draft just to show you and made test only for blackrock.
need to propagate anywhere.

**Comment by samuelgarcia** [comment]
@h-mayorquin : here in progress PR to propagate more annotation and array_annotation (aka properties in SI) in recording extractor.
This relate an old discuss on metadata we had 2 weeks ago.

**Comment by h-mayorquin** [comment]
> @h-mayorquin : here in progress PR to propagate more annotation and array_annotation (aka properties in SI) in recording extractor. This relate an old discuss on metadata we had 2 weeks ago.

Nice, so the neoextractors will have an extra property `all_annotations` that allows the propagation.

**Comment by samuelgarcia** [comment]
Yep exactly : you can test it.

**Comment by samuelgarcia** [comment]
@alejoe91 : test are passing now.
MAybe we can merge.

**Review by alejoe91 — APPROVED** [review]

**Comment by yger** [comment]
Maybe this is related to this PR, but I have bugs with locs_2d and stream_id after updating to master....

**Comment by samuelgarcia** [comment]
Too bad. I should not have removed this `locs_2d` it is a mistake from me.
It will raise bug anyware.
Sorry.
I will put it back.
@alejoe91 : do you remember why ?

**Pull request merged**